packages/table/src/TableBody.tsx

```diff
@@ -1,17 +1,58 @@
-import React from 'react';
-import { useTableContext } from './TableContext';
+import React, { useMemo, useCallback } from 'react';
+import { getDataComparisonFunction, useTableContext } from './TableContext';
 import { TableProps } from './Table';
 
 type TableBodyProps<Shape> = Pick<TableProps<Shape>, 'children'>;
 
+/**
+ * Hook that avoids re-rendering children when the data used to
+ * render them hasn't changed, even if their ordering has changed.
+ */
+function useRenderedChildren<Datum>(
+  data: Array<Datum>,
+  renderFunction: React.FunctionComponent<Datum>,
+  compareFn?: (a: Datum, b: Datum) => number,
+): Array<React.ReactNode> {
+  const resultMap = useMemo(() => {
+    const resultMap = new Map<Datum, React.ReactNode>();
+    data.forEach((datum, index) =>
+      resultMap.set(
+        datum,
+        <React.Fragment key={index}>{renderFunction(datum)}</React.Fragment>,
+      ),
+    );
+    return resultMap;
+  }, [data, renderFunction]);
+
+  return useMemo(() => {
+    if (!compareFn) {
+      return Array.from(resultMap.values());
+    }
+
+    return [...data].sort(compareFn).map(datum => resultMap.get(datum));
+  }, [data, resultMap, compareFn]);
+}
+
 function TableBody<Shape>({ children }: TableBodyProps<Shape>) {
   const {
-    state: { data },
+    state: { data, sort },
```

Comment:
Random thought - would we gain anything moving sort to its own context?

Reply:
I checked and yeah, `Row` subscribes to table context and doesn't use `sort`, so that's a lot of re-renders we can prevent.